src/dynamic-routes/path-parser.ts

```diff
@@ -0,0 +1,13 @@
+import {MemoryRouter, UrlObject, UrlObjectWithPath} from "../MemoryRouter";
+
+declare module "../MemoryRouter" {
+  interface MemoryRouter {
+    setPathParser(parser: (url: UrlObject) => UrlObjectWithPath): void
+  }
+}
+
+MemoryRouter.prototype.setPathParser = function(parser: (url: UrlObject) => UrlObjectWithPath) {
```

Comment:
Since this file is so simple, I think it makes sense to include it in the `MemoryRouter` file itself.  "PathParser" isn't necessarily tied to the `dynamic-routes`.

Reply:
That's reasonable enough; my rationale here was that since the pathParser isn't really pertinent outside the context of dynamic routes, it might as well not be exposed unless the user of the library opts in to route parsing. I'm happy to move it back if you think it has some value outside of that context.

Reply:
@scottrippey let me know if this is something you'd like me to address prior to merging; if not, I am feeling pretty good about where this PR is at.

Reply:
No, this is totally fine.